enumerable.rb

```diff
@@ -0,0 +1,112 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    for x in self
+      yield(x)
+    end
+    x
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    x = 0
+    for y in self
+      yield(x, y)
```

Comment:
- [ ] In the each_with_index the element comes first and the index second. If x is your index it should be the second param.